gwcs/spectroscopy.py

```diff
@@ -192,3 +193,216 @@ def input_units(self):
             return {'wavelength': 1 / self.groove_density.unit,
                     'alpha_in': u.Unit(1),
                     'beta_in': u.Unit(1)}
+
+
+class Snell3D(Model):
+    """
+    Snell model in 3D form.
+
+    Inputs are direction cosines.
```

Comment:
The docstring should be updated to indicate the additional input.

Reply:
Done